oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/NodeImpl.java

```diff
@@ -1379,6 +1380,11 @@ private Property internalSetProperty(
         final String oakName = getOakPathOrThrow(checkNotNull(jcrName));
         final PropertyState state = createSingleState(
                 oakName, value, Type.fromTag(value.getType(), false));
+        long maxStringPropertyLength = sessionContext.getRepository().getDescriptorValue(RepositoryImpl.MAX_STRING_PROPERTY_SIZE).getLong();
```

Comment:
internalSetProperty is called very, very often. It is performance critical.

The method call getDescriptorValue is relatively slow. Instead, the value should be stored in the session context or the repository as a field, so it is faster to retrieve the value.

Also, the value is only needed for properties of type "string". So this line can be moved inside the "if (type == "string")".